File: php-sdk/src/Models/ApiResponse.php
<?php

declare(strict_types=1);

namespace Cleanster\Models;

/**
 * Standard response wrapper returned by every SDK method.
 *
 * @property int    $status  HTTP-style status code (e.g., 200).
 * @property string $message Human-readable status (e.g., "OK").
 * @property mixed  $data    Typed payload — a model object, array of models, or raw array.
 */
final class ApiResponse
{
    public function __construct(
        public readonly int    $status,
        public readonly string $message,
        public readonly mixed  $data,
    ) {}
}

File: csharp-sdk/src/Cleanster/Models/ApiResponse.cs
namespace Cleanster.Models;

/// <summary>
/// Standard response wrapper returned by every SDK method.
/// </summary>
/// <typeparam name="T">The type of the <see cref="Data"/> payload.</typeparam>
/// <param name="Status">HTTP-style status code (e.g., 200).</param>
/// <param name="Message">Human-readable status string (e.g., "OK").</param>
/// <param name="Data">Typed payload — a model object, list of models, or raw JsonElement.</param>
public sealed record ApiResponse<T>(int Status, string Message, T Data);
